src/main/java/seedu/financialplanner/cashflow/Expense.java

```diff
@@ -40,8 +41,15 @@ public IncomeType getIncomeType() {
         return null;
     }
 
-    private void addExpenseValue() {
-        balance -= this.amount;
+    private void addExpenseValue() throws FinancialPlannerException {
+        double tempBalance = balance - this.amount;
+
+        if (tempBalance < -MAX_AMOUNT) {
+            throw new FinancialPlannerException("Balance exceeded minimum value this program can hold.");
```

Comment:
Should tell the user to re-add another expense as this expense would not be recorded